src/main/java/core/basesyntax/SqrtOperation.java

```diff
@@ -1,13 +1,19 @@
 package core.basesyntax;
 
+import java.io.IOException;
+
 /**
  * <p>Реализуйте метод sqrt(), вычисляющий квадратный корень числа. В отличие от Math.sqrt(),
  * это метод при передаче отрицательного параметра должен бросать исключение
  * java.lang.IllegalArgumentException с сообщением "Expected non-negative number, got ?",
  * где вместо вопросика будет подставлено фактически переданное в метод число.</p>
  */
 public class SqrtOperation {
-    public double sqrt(double number) {
-        return 0.0;
+    public double sqrt(double number) throws IllegalArgumentException {
+        if (number < 0) {
+            throw new IllegalArgumentException("Expected non-negative number, got " + number);
+        } else {
```

Comment:
why you are using else? do you know how if statement works? what happens when you are using return or throwing the exception inside the if?

Reply:
No need in else here... if() switches its code if condition is true, if false - switches the code below it.